Renovate stopped working for Kotlin Gradle projects with buildSrc folder

**What Renovate type are you using?**

Tested on both Whitesource Renovate (19.211.3) and a private hosted version (19.212.1)

**Describe the bug**

Renovate has stopped working for Kotlin Gradle projects that use the kotlin-dsl plugin (enables the [buildSrc](https://docs.gradle.org/current/userguide/organizing_gradle_projects.html#sec:build_sources) folder). After checking the logs I believe this has been working fine until sometime today and I suspect it has something to do with https://github.com/renovatebot/renovate/pull/5975 (version 19.211.2).

You can see in the errors that it's unable to run gradle because it tries to compile `buildSrc` and it can't find the JDK:
```
"errMessage": "Command failed: docker run --rm --name=renovate_gradle --label=renovate_child -v \"/mnt/renovate/gh/ps-feng/renovate-gradle-kts-bug\":\"/mnt/renovate/gh/ps-feng/renovate-gradle-kts-bug\" -v \"/tmp/renovate-cache\":\"/tmp/renovate-cache\" -w \"/mnt/renovate/gh/ps-feng/renovate-gradle-kts-bug\" renovate/gradle bash -l -c \"./gradlew --init-script renovate-plugin.gradle renovate\"\n\nFAILURE: Build failed with an exception.\n\n* What went wrong:\nExecution failed for task ':buildSrc:compileKotlin'.\n> Kotlin could not find the required JDK tools in the Java installation '/usr/lib/jvm/java-8-openjdk-amd64/jre' used by Gradle. Make sure Gradle is running on a JDK, not JRE.\n\n* Try:\nRun with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.\n\n* Get more help at https://help.gradle.org\n\nBUILD FAILED in 3m 4s\n"
```
I'm not sure why this would fail now and not earlier because when it was depending on Java 11 it was still JRE.

**Did you see anything helpful in debug logs?**
These are the logs for Renovate job #165690777, corresponding to this [sample repository](https://github.com/ps-feng/renovate-gradle-kts-bug).

<details>

```
INFO: Repository started
{
  "renovateVersion": "19.211.3"
}
DEBUG: Using localDir: /mnt/renovate/gh/ps-feng/renovate-gradle-kts-bug
DEBUG: initRepo("ps-feng/renovate-gradle-kts-bug")
DEBUG: No dangling containers to remove
DEBUG: ps-feng/renovate-gradle-kts-bug owner = ps-feng
DEBUG: ps-feng/renovate-gradle-kts-bug default branch = master
DEBUG: Using app token for git init
DEBUG: Initializing git repository into /mnt/renovate/gh/ps-feng/renovate-gradle-kts-bug
DEBUG: git clone completed
{
  "seconds": 0.7
}
DEBUG: latest commit
{
  "latestCommitDate": "2020-04-17 17:35:29 +0200"
}
DEBUG: Git private key configured, but not being set
DEBUG: Setting git author
{
  "gitAuthor": {
    "name": "Renovate Bot",
    "email": "bot@renovateapp.com"
  }
}
DEBUG: resetMemCache()
DEBUG: detectSemanticCommits()
DEBUG: getCommitMessages
DEBUG: Semantic commits detection: unknown
DEBUG: No semantic commits detected
DEBUG: checkOnboarding()
DEBUG: isOnboarded()
DEBUG: findFile(renovate.json)
DEBUG: config file exists
DEBUG: ensureIssueClosing(Action required: Add a Renovate config)
DEBUG: Retrieving issueList
DEBUG: Retrieved 0 issues
DEBUG: Repo is onboarded
DEBUG: Found renovate.json config file
DEBUG: Repository config
{
  "configFile": "renovate.json",
  "config": {
    "extends": [
      "config:base"
    ]
  }
}
DEBUG: migrateAndValidate()
DEBUG: No config migration necessary
DEBUG: massaged config
{
  "config": {
    "extends": [
      "config:base"
    ]
  }
}
DEBUG: migrated config
{
  "config": {
    "extends": [
      "config:base"
    ]
  }
}
DEBUG: Found repo ignorePaths
{
  "ignorePaths": [
    "**/node_modules/**",
    "**/bower_components/**",
    "**/vendor/**",
    "**/examples/**",
    "**/__tests__/**",
    "**/test/**",
    "**/tests/**",
    "**/__fixtures__/**"
  ]
}
DEBUG: checkBaseBranch()
DEBUG: config.repoIsOnboarded=true
DEBUG: Setting baseBranch to master
DEBUG: latest commit
{
  "branchName": "master",
  "latestCommitDate": "2020-04-17 17:35:29 +0200"
}
DEBUG: Setting branchPrefix: renovate/
DEBUG: No vulnerability alerts found
DEBUG: processRepo()
DEBUG: No baseBranches
DEBUG: extractAndUpdate()
DEBUG: Using file match: (^|/)tasks/[^/]+\.ya?ml$ for manager ansible
DEBUG: Using file match: (^|/)requirements\.ya?ml$ for manager ansible-galaxy
DEBUG: Using file match: (^|/)WORKSPACE$ for manager bazel
DEBUG: Using file match: \.bzl$ for manager bazel
DEBUG: Using file match: buildkite\.ya?ml for manager buildkite
DEBUG: Using file match: \.buildkite/.+\.ya?ml$ for manager buildkite
DEBUG: Using file match: (^|/)Gemfile$ for manager bundler
DEBUG: cargo is disabled
DEBUG: Using file match: (^|/).circleci/config.yml$ for manager circleci
DEBUG: Using file match: (^|/)Podfile$ for manager cocoapods
DEBUG: Using file match: (^|/)([\w-]*)composer.json$ for manager composer
DEBUG: Using file match: (^|/)deps\.edn$ for manager deps-edn
DEBUG: Using file match: (^|/)docker-compose[^/]*\.ya?ml$ for manager docker-compose
DEBUG: Using file match: (^|/|\.)Dockerfile$ for manager dockerfile
DEBUG: Using file match: (^|/)Dockerfile\.[^/]*$ for manager dockerfile
DEBUG: Using file match: (^|/).drone.yml$ for manager droneci
DEBUG: git-submodules is disabled
DEBUG: github-actions is disabled
DEBUG: Using file match: ^\.gitlab-ci\.yml$ for manager gitlabci
DEBUG: Using file match: ^\.gitlab-ci\.yml$ for manager gitlabci-include
DEBUG: Using file match: (^|/)go.mod$ for manager gomod
DEBUG: Using file match: \.gradle(\.kts)?$ for manager gradle
DEBUG: Using file match: (^|/)gradle.properties$ for manager gradle
DEBUG: Matched 4 file(s) for manager gradle: build.gradle.kts, buildSrc/build.gradle.kts, settings.gradle, gradle.properties
DEBUG: Extracting dependencies from all gradle files
DEBUG: Creating renovate-plugin.gradle file with renovate gradle plugin
DEBUG: Start gradle command
{
  "cmd": "./gradlew --init-script renovate-plugin.gradle renovate"
}
DEBUG: Using docker to execute
DEBUG: Executing command
{
  "command": "docker run --rm --name=renovate_gradle --label=renovate_child -v \"/mnt/renovate/gh/ps-feng/renovate-gradle-kts-bug\":\"/mnt/renovate/gh/ps-feng/renovate-gradle-kts-bug\" -v \"/tmp/renovate-cache\":\"/tmp/renovate-cache\" -w \"/mnt/renovate/gh/ps-feng/renovate-gradle-kts-bug\" renovate/gradle bash -l -c \"./gradlew --init-script renovate-plugin.gradle renovate\""
}
WARN: Gradle extraction failed
{
  "errMessage": "Command failed: docker run --rm --name=renovate_gradle --label=renovate_child -v \"/mnt/renovate/gh/ps-feng/renovate-gradle-kts-bug\":\"/mnt/renovate/gh/ps-feng/renovate-gradle-kts-bug\" -v \"/tmp/renovate-cache\":\"/tmp/renovate-cache\" -w \"/mnt/renovate/gh/ps-feng/renovate-gradle-kts-bug\" renovate/gradle bash -l -c \"./gradlew --init-script renovate-plugin.gradle renovate\"\n\nFAILURE: Build failed with an exception.\n\n* What went wrong:\nExecution failed for task ':buildSrc:compileKotlin'.\n> Kotlin could not find the required JDK tools in the Java installation '/usr/lib/jvm/java-8-openjdk-amd64/jre' used by Gradle. Make sure Gradle is running on a JDK, not JRE.\n\n* Try:\nRun with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.\n\n* Get more help at https://help.gradle.org\n\nBUILD FAILED in 3m 4s\n"
}
DEBUG: Using file match: (^|/)gradle/wrapper/gradle-wrapper.properties$ for manager gradle-wrapper
DEBUG: Matched 1 file(s) for manager gradle-wrapper: gradle/wrapper/gradle-wrapper.properties
DEBUG: gradle-wrapper.extractPackageFile()
DEBUG: Gradle Wrapper(depType="gradle-wrapper")
{
  "datasource": "gradle-version",
  "depName": "gradle",
  "currentValue": "6.3.0",
  "managerData": {
    "lineNumber": 2,
    "gradleWrapperType": "bin"
  },
  "versioning": "semver"
}
DEBUG: Found gradle-wrapper package files
DEBUG: Using file match: (^|/)requirements.yaml$ for manager helm-requirements
DEBUG: Using file match: (^|/)values.yaml$ for manager helm-values
DEBUG: Using file match: (^|/)helmfile.yaml$ for manager helmfile
DEBUG: Using file match: ^Formula/[^/]+[.]rb$ for manager homebrew
DEBUG: Using file match: \.html?$ for manager html
DEBUG: Using file match: (^|/)kustomization\.yaml for manager kustomize
DEBUG: Using file match: (^|/)project\.clj$ for manager leiningen
DEBUG: Using file match: \.pom\.xml$ for manager maven
DEBUG: Using file match: (^|/)pom\.xml$ for manager maven
DEBUG: Using file match: (^|/)package.js$ for manager meteor
DEBUG: Using file match: (^|/)mix\.exs$ for manager mix
DEBUG: Using file match: ^.node-version$ for manager nodenv
DEBUG: Using file match: (^|/)package.json$ for manager npm
DEBUG: Using file match: \.(?:cs|fs|vb)proj$ for manager nuget
DEBUG: Using file match: ^.nvmrc$ for manager nvm
DEBUG: Using file match: (^|/)([\w-]*)requirements.(txt|pip)$ for manager pip_requirements
DEBUG: Using file match: (^|/)setup.py$ for manager pip_setup
DEBUG: pipenv is disabled
DEBUG: Using file match: (^|/)pyproject\.toml$ for manager poetry
DEBUG: Using file match: (^|/)pubspec\.ya?ml$ for manager pub
DEBUG: Using file match: (^|/)\.ruby-version$ for manager ruby-version
DEBUG: Using file match: \.sbt$ for manager sbt
DEBUG: Using file match: project/[^/]*.scala$ for manager sbt
DEBUG: Using file match: (^|/)Package\.swift for manager swift
DEBUG: Using file match: \.tf$ for manager terraform
DEBUG: Using file match: ^.travis.yml$ for manager travis
DEBUG: Found 1 package file(s)
INFO: Extraction statistics
{
  "stats": {
    "managers": {
      "gradle-wrapper": {
        "fileCount": 1,
        "depCount": 1
      }
    },
    "fileCount": 1,
    "depCount": 1
  }
}
DEBUG: fetchUpdates complete
DEBUG: packageFiles with updates
{
  "config": {
    "gradle-wrapper": [
      {
        "packageFile": "gradle/wrapper/gradle-wrapper.properties",
        "manager": "gradle-wrapper",
        "deps": [
          {
            "datasource": "gradle-version",
            "depType": "gradle-wrapper",
            "depName": "gradle",
            "currentValue": "6.3.0",
            "managerData": {
              "lineNumber": 2,
              "gradleWrapperType": "bin"
            },
            "versioning": "semver",
            "depIndex": 0,
            "updates": [],
            "warnings": [],
            "sourceUrl": "https://github.com/gradle/gradle",
            "homepage": "https://gradle.org"
          }
        ]
      }
    ]
  }
}
DEBUG: branchifyUpgrades
DEBUG: 0 flattened updates found:
DEBUG: Returning 0 branch(es)
DEBUG: config.repoIsOnboarded=true
DEBUG: Processing 0 branches:
DEBUG: Calculating hourly PRs remaining
DEBUG: Retrieving PR list
DEBUG: Retrieved 1 Pull Requests
DEBUG: currentHourStart=1587135600000
DEBUG: PR hourly limit remaining: 2
DEBUG: Enforcing prConcurrentLimit (20)
DEBUG: 0 PRs are currently open
DEBUG: PR concurrent limit remaining: 20
DEBUG: branchPrefix: renovate/
DEBUG: Found 0 Renovate PRs
{
  "renovatePrs": []
}
DEBUG: Removing any stale branches
DEBUG: config.repoIsOnboarded=true
DEBUG: No renovate branches found
DEBUG: ensureIssueClosing(Action Required: Fix Renovate Configuration)
INFO: Repository finished
```

</details>

<!-- If you're running self-hosted, run with `--log-level=debug` or LOG_LEVEL=debug and search for whatever dependency/branch/PR that is causing the problem. If you are using the Renovate App, log into https://app.renovatebot.com/dashboard and locate the correct job log for when the problem occurred (e.g. when the PR was created). The Job ID will help us locate it. -->

**To Reproduce**

<!-- To fix a bug, we nearly always need a *minimal* repo to reproduce it in, before verifying that our fix works using the same repo. If you provide a public repo that already reproduces the problem, then your bug will get highest priority for fixing. If you can't reproduce it in a simple repo, do your best to describe how it could be reproduced, or under what circumstances the bug occurs. -->

1. Create a Kotlin Gradle project with `buildSrc`.
2. Run Renovate bot on it

See this sample repository: https://github.com/ps-feng/renovate-gradle-kts-bug

**Additional context**

<!-- Add any other context about the problem here, including your own debugging or ideas on what went wrong. -->
We could maybe try including the JDK instead of just the JRE.

--- a/Dockerfile
+++ b/Dockerfile
@@ -68,7 +68,7 @@ ENV RENOVATE_BINARY_SOURCE=docker
 FROM final-base as latest
 
 RUN apt-get update && \
-    apt-get install -y gpg wget unzip xz-utils openssh-client bsdtar build-essential openjdk-8-jre-headless dirmngr && \
+    apt-get install -y gpg wget unzip xz-utils openssh-client bsdtar build-essential openjdk-8-jdk-headless dirmngr && \
     rm -rf /var/lib/apt/lists/*
 
 

